static/js/dashboard-spa.js

```diff
@@ -0,0 +1,538 @@
+      }
+    },
+
+    async handleNoteChange(alias) {
+      try {
+        const res = await fetch(`/api/aliases/${alias.id}`, {
+          method: "PUT",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({
+            note: alias.note,
+          }),
+        });
+
+        if (res.ok) {
+          toastr.success(`Note saved for ${alias.email}`);
+        } else {
+          throw new Error("Note could not be saved");
+        }
+
+      } catch (err) {
+        toastr.error("Sorry for the inconvenience! Could you refresh the page & retry please?", "Note could not be saved");
+      }
+    },
+
+    async handleDisplayNameChange(alias) {
+      try {
+        let res = await fetch(`/api/aliases/${alias.id}`, {
+          method: "PUT",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({
+            name: alias.name,
+          }),
+        });
+
+        if (res.ok) {
+          toastr.success(`Display name saved for ${alias.email}`);
+        } else {
+          throw new Error("Could not save Display name");
+        }
+
+      } catch (err) {
+        toastr.error("Sorry for the inconvenience! Could you refresh the page & retry please?", "Could not save Display name")
+      }
+
+    },
+
+    async handlePgpToggle(alias) {
+      try {
+        let res = await fetch(`/api/aliases/${alias.id}`, {
+          method: "PUT",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({
+            disable_pgp: alias.disable_pgp,
+          }),
+        });
+
+        if (res.ok) {
+          toastr.success(`PGP ${alias.disable_pgp ? "disabled" : "enabled"} for ${alias.email}`);
+        } else {
+          throw new Error("Could not toggle PGP")
+        }
+
+      } catch (err) {
+        alias.disable_pgp = !alias.disable_pgp;
+        toastr.error("Sorry for the inconvenience! Could you refresh the page & retry please?", "Could not toggle PGP");
+      }
+    },
+
+    async handlePin(alias) {
+      try {
+        let res = await fetch(`/api/aliases/${alias.id}`, {
+          method: "PUT",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({
+            pinned: !alias.pinned,
+          }),
+        });
+
+        if (res.ok) {
+          alias.pinned = !alias.pinned;
+          if (alias.pinned) {
+            // make alias appear at the top of the alias list
+            const index = this.aliasesArray.findIndex((a) => a.id === alias.id);
+            this.aliasesArray.splice(index, 1);
+            this.aliasesArray.unshift(alias);
+            toastr.success(`${alias.email} is pinned`);
+          } else {
+            // unpin: make alias appear at the bottom of the alias list
```

Comment:
Though moving a pinned alias to the top is a good idea, handling the reverse case when an alias is unpinned is much harder. I think let's just not handle this for now and leave the item where it is :)